certbot/tests/error_handler_test.py

```diff
@@ -30,25 +50,20 @@ def test_context_manager(self):
         self.init_func.assert_called_once_with(*self.init_args,
                                                **self.init_kwargs)
 
-    @mock.patch('certbot.error_handler.os')
-    @mock.patch('certbot.error_handler.signal')
-    def test_signal_handler(self, mock_signal, mock_os):
-        # pylint: disable=protected-access
-        mock_signal.getsignal.return_value = signal.SIG_DFL
-        self.handler.set_signal_handlers()
-        signal_handler = self.handler._signal_handler
-        for signum in self.signals:
-            mock_signal.signal.assert_any_call(signum, signal_handler)
-
-        signum = self.signals[0]
-        signal_handler(signum, None)
+    def test_context_manager_with_signal(self):
+        with signal_receiver(self.signals) as signals_received:
+            with self.handler:
+                should_be_42 = 42
+                send_signal(signal.SIGTERM)
```

Comment:
I think we should avoid hardcoding specific signals in these tests. Why?

1. When changing the signals list in `error_handler.py` we have to remember to update these as well.
2. In the case of `SIGHUP` below, we don't actually use that signal on all systems

Instead, I think we should use `self.signals[0]`, `self.signals[-1]`, etc.